apps/scan/frontend/src/components/misvote_warnings/use_layout_config_hook.test.tsx

```diff
@@ -0,0 +1,168 @@
+import { SizeMode } from '@votingworks/types';
+import { render } from '../../../test/react_testing_library';
+import { Layout, MisvoteWarningsProps } from './types';
+import { useLayoutConfig } from './use_layout_config_hook';
+import { CONFIG } from './constants';
+import { generateContests } from './test_utils.test';
+
+let hookResult: Layout | null = null;
+
+function TestComponent(props: MisvoteWarningsProps) {
+  hookResult = useLayoutConfig(props);
+
+  return <div>foo</div>;
+}
+
+beforeEach(() => {
+  hookResult = null;
+});
+
+test('varies according to size mode', () => {
+  const props: MisvoteWarningsProps = {
+    blankContests: generateContests(2),
+    overvoteContests: generateContests(1),
+    partiallyVotedContests: generateContests(3),
+  };
+
+  render(<TestComponent {...props} />, { vxTheme: { sizeMode: 'xl' } });
+  const layoutXl = hookResult;
+
+  render(<TestComponent {...props} />, { vxTheme: { sizeMode: 'm' } });
+  const layoutM = hookResult;
+
+  expect(layoutXl).not.toEqual(layoutM);
+});
+
+test('sets numCardsPerRow appropriately', () => {
+  const sizeMode: SizeMode = 'm';
+  const config = CONFIG[sizeMode];
+
+  const { rerender } = render(
+    <TestComponent
+      blankContests={generateContests(1)}
+      overvoteContests={generateContests(1)}
+      partiallyVotedContests={generateContests(1)}
+    />,
+    { vxTheme: { sizeMode } }
+  );
+  expect(hookResult).toEqual(
+    expect.objectContaining<Partial<Layout>>({
+      numCardsPerRow: config.maxCardsPerRow,
+    })
+  );
+
+  rerender(
+    <TestComponent
+      blankContests={generateContests(1)}
+      overvoteContests={[]}
+      partiallyVotedContests={[]}
+    />
+  );
+  expect(hookResult).toEqual(
+    expect.objectContaining<Partial<Layout>>({
+      numCardsPerRow: 1,
+    })
+  );
+});
```

Comment:
It looks like maybe you separated the layout config as a hook for test-ability? That seems fine.

Maybe I'm misremembering, but I think the only data actually used is the `showPreview` flag so I don't know if you need to assert against these other things / have them in there.

Reply:
yeah, the other fields are referenced in `warning_details`, which was the initial motivation for extracting to a common util, but it definitely made testing easier as well!